_posts/2024-06-07-opensearch-performance-2.14.md

```diff
@@ -0,0 +1,161 @@
+
+- Improving search relevancy by supporting reranking on compressed vectors.
+- Supporting ANN vector search for the multi-tenant case, where each tenant has its own subcollection. This is especially valuable for customers each of whose user organizations has a large vector collection.
+
+### Performance benchmarking
+
+We are planning to update [the Big5 workload](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/big5) mappings and settings. In particular, we'll add support for dynamic templates, configurable merge policies, match-only text, and setting the default query field as `message`.
+
+## Conclusion
+
+While we're continuing to expand OpenSearch functionality, we are also investing in improving the performance, efficiency, and robustness of the system for every workload. In particular, vector techniques are being widely adopted both for pure semantic search and for hybrid semantic and lexical search.
+
+The OpenSearch team at AWS works in collaboration with the larger OpenSearch community. Without its contributions to testing, feedback, and development, OpenSearch would not be where it is today.
+
+Stay tuned to our blog and GitHub for continuous updates and insights into our progress and future plans.
+
+## Appendix: Benchmark tests and results
+
+This section presents details on the performance benchmarks we used and the results we obtained. The data was collected using OpenSearch Benchmark to run the Big5 workload against different distributions and configurations of OpenSearch. In particular:
+
+- We used OpenSearch Benchmark---our standard benchmarking tool---for testing. Rally might produce different results.
+- We ran each test against a single-node cluster to ensure better reproducibility and easier setup.
+- The instance type was *c5.2xlarge*: 8 vCPU, 16 GB, an average instance type. Using oversized instances can conceal resource usage efficiency improvements.
+- The Big5 index was comprised of a single shard, no replicas (run with `--workload-params="number_of_replicas:0"`).
+- During the test, we ingested data using a single bulk indexing client to ensure that data is ingested in chronological order.
+- We ran the tests in [benchmarking mode](https://opensearch.org/docs/latest/benchmark/user-guide/target-throughput/#benchmarking-mode) (`target-throughput` was disabled) so that the OpenSearch client sends requests to the OpenSearch cluster as quickly as possible.
+- We configured the indexing Log Structured Merge (LSM) with LogByteSizeMergePolicy because this optimization was in place last year, prior to the blog post being published.
+- The corpus size of the workload was 60 GB, 70M documents. The store size after ingestion was 15 GB for the primary shard. If overhead like doc values is eliminated, Resident Set Size (RSS) should be about 8 GB, which should match the JVM heap size for the instance. Having most data resident in memory should provide a good assessment of performance improvements.
```

Comment:
Last sentence: Should "in-memory" be hyphenated? Also, the last sentence is a bit "fluffy." Can we simplify and say something like "Having most data in-memory should provide an ideal performance assessment"?

Reply:
Agree with - Having most data in-memory should provide an ideal performance assessment